Classes/MapEditor/MapEditor.cs

```diff
@@ -474,8 +470,32 @@ public class MapEditor {
     public void MirrorSelection() {
         TransformSelection(NoteTransforms.Mirror());
     }
-    public void ShiftSelectionByBeat(double beat) {
-        TransformSelection(NoteTransforms.RowShift(beat));
+    public void ShiftSelectionByBeat(MoveNote direction) {
+        if (currentMapDifficulty == null) {
+            return;
+        }
+
+        List<Note> movedNotes = currentMapDifficulty.selectedNotes
```

Comment:
I don't like replacing ```TransformSelection``` and keeping the logic in here - it should stay in ```NoteTransforms```. The same actually goes for ```QuantizeSelection```.

I think both should be implemented as parametrized ```NoteTransforms```, injecting ```GetLastBeatChange``` and ```GetGridLength``` as parameters of type ```Func<Double, BPMChange>``` and ```Func<Double, Integer, Double>``` respectively, to resolve the issue of not having access to those, instead of moving the logic to ```MapEditor```.

Reply:
I'm not sure what the point of this architecture is, what is the advantage of outsourcing this into TransformSelection?